src/main/java/org/junit/runner/OrderWith.java

```diff
@@ -0,0 +1,26 @@
+package org.junit.runner;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.runner.manipulation.Ordering;
+
+/**
+ * When a test class is annotated with <code>&#064;OrderWith</code> or extends a class annotated
+ * with <code>&#064;OrderWith</code>, JUnit will order the tests in the test class (and child
+ * test classes, if any) using the ordering defined by the {@link Ordering} class.
+ *
+ * @since 4.13
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface OrderWith {
```

Comment:
2) Changes like this (which affect how _any_ class processed by JUnit is run, regardless of the chosen runner) have been rare, on purpose.  There's a good argument to be made in favor of this one, but as long as we've got the hood open, I wonder if something more general would make sense.

For example, there's not currently a FilterWith annotation, but once we accept this change, that seems to be a natural next square in the matrix to fill in, which would be _another_ top-level change.  Also related would be the common request to attach a custom runlistener to a class, ListenWith

Just thinking out loud, what if we made a top-level annotation (RunRule?  Modify?) which would handle any of these changes?  Essentially:

public interface RunRule {
  public Runner apply(Runner runner, Description classBeingRun);
}

and then:

public class MyTest {
  @UseRunRule public static RunRule ordering = new UseOrdering(Ordering.random());
  @UseRunRule public static RunRule listening = new AddListener(new MyListener());
  // etc...
}

Note that: 

a) This builds on the static field suggestion in (1), but could be orthogonal to it.
b) I'm not thrilled with the top-of-my-head naming here; especially, RunRule makes this sound tied to Rule, for better or for worse.